retriever/bin/retriever.js

```diff
@@ -21,21 +21,19 @@ export default {
     try {
       return await this._fetch(request, env, ctx, retrieveFile)
     } catch (error) {
-      console.error(error)
-      const message =
-        typeof error === 'object' &&
-        error !== null &&
-        'message' in error &&
-        typeof error.message === 'string'
-          ? error.message
-          : String(error)
-      const status =
+      const errHasStatus =
         typeof error === 'object' &&
         error !== null &&
         'status' in error &&
         typeof error.status === 'number'
-          ? error.status
-          : 500
+      const message =
+        errHasStatus && 'message' in error && typeof error.message === 'string'
```

Comment:
Internal server errors (5xx) can contain sensitive information (e.g. "connection refused" often includes the hostname of the remote server); therefore, we shouldn't be exposing the error message to the clients.

On the other hand, I can see an argument that if the caller specified `error.status = 5xx` then they probably specified also a custom error message that should be safe to send to the clients. (But this will be difficult to enforce in practice.)

WDYT?

```suggestion
        errHasStatus && error.status < 500 && 'message' in error && typeof error.message === 'string'
```